crates/bevy_reflect/src/map.rs

```diff
@@ -246,6 +252,18 @@ impl Map for DynamicMap {
         }
     }
 
+    fn remove(&mut self, key: &dyn Reflect) -> Option<Box<dyn Reflect>> {
+        match self.indices.entry(key.reflect_hash().expect(HASH_ERROR)) {
+            Entry::Occupied(entry) => {
+                let index = *entry.get();
+                entry.remove();
+                let (_key, value) = self.values.remove(index);
+                Some(value)
+            },
+            Entry::Vacant(_) => None,
+        }
```

Comment:
Nit: Could this be simplified by doing something like
```suggestion
        let index = self.indices.remove(key.reflect_hash().expect(HASH_ERROR))?;
        let (_key, value) = self.values.remove(index);
        Some(value)
```